dns_discover.go

```diff
@@ -40,6 +41,20 @@ func discoverDNS(domain string, port int) (servers []string, ttl time.Duration,
 	return
 }
 
+// retryingFindTXT will, on any DNS failure, retry for up to 10 minutes before
+// giving up and returning an empty []string of records
+func retryingFindTXT(fqdn string) (records []string, ttl time.Duration, err error) {
+	err = backoff.Retry(
+		func() error {
+			records, ttl, err = findTXT(fqdn)
+			if err != nil {
+				log.Error("Retrying DNS query. Query failed with: %s", err.Error())
+			}
+			return err
+		}, backoff.NewExponentialBackOff())
+	return
```

Comment:
> Finally, in some cases you need to name a result parameter in order to change it in a deferred closure. That is always okay.
-- [Go Style Guide](https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Named_Result_Parameters)

I think given how short the wrapper is there's little chance of confusion.